Add migration guide for astro (#1248)
clerk/clerk-docs: docs/references/astro/migrating-from-astro-community-sdk.mdx

---
title: 'Migrating from the Astro community SDK'
description: Learn how to migrate from the Astro community SDK to the Clerk Astro SDK.
---

# Migrating from the Astro community SDK

> In June 2024, we introduced official support for Astro. This minimal guide highlights the breaking changes that were introduced and provides examples on how to resolve them.

Reviewer: ```suggestion
In July 2024, we introduced official support for Astro. This migration guide covers converting from the `astro-clerk-auth` community SDK to the official SDK. It covers the breaking changes that were introduced and provides examples on how to resolve them.
```

**Overview:**
- Install `@clerk/astro`, uninstall `astro-clerk-auth`
- Update import paths

> - Replace already deprecated APIs

Reviewer: ```suggestion
- Replace deprecated APIs
```

## Installation

Uninstall the community SDK and install the new offical SDK for Astro.

<CodeBlockTabs type="installer" options={["npm", "yarn", "pnpm"]}>
    ```bash filename="terminal"
    npm uninstall astro-clerk-auth
    npm install @clerk/astro
    ```

    ```bash filename="terminal"
    yarn remove astro-clerk-auth
    yarn add @clerk/astro
    ```

    ```bash filename="terminal"
    pnpm remove astro-clerk-auth
    pnpm add @clerk/astro
    ```
</CodeBlockTabs>

## Update import paths
We have kept breaking changes to the minimum and there is a great chance that it will not affect your existing codebase.

> As a quick solution find all the usages of `astro-clerk-auth/` and replace them `@clerk/astro/`

Reviewer: ```suggestion
As a quick solution find all the usages of `astro-clerk-auth` and replace them `@clerk/astro`
```
I think this reads a little nicer, with the same effect. Not a required change though.

```diff
- import { clerkMiddleware } from "astro-clerk-auth/server"
+ import { clerkMiddleware } from "@clerk/astro/server"

- import { SignIn } from "astro-clerk-auth/components/interactive"
+ import { SignIn } from "@clerk/astro/components/interactive"

```

## Breaking Changes

### The integration

When using the integration from `@clerk/astro` we are now hotloading clerk-js by default.

```diff
import { defineConfig } from "astro/config";
import node from "@astrojs/node";
- import clerk from "astro-clerk-auth/hotload";
+ import clerk from "@clerk/astro";

export default defineConfig({
    integrations: [clerk()],
    output: "server",
    adapter: node({ mode: "standalone" }),
});
```

**Bundled** \
And in order to opt-in to the bundled clerk-js strategy you can do the following

> ```diff

Reviewer: ```suggestion
```filename="astro.config.mjs" {2-3}
```

import { defineConfig } from "astro/config";
import node from "@astrojs/node";
- import clerk from "astro-clerk-auth";

> + import clerk from "@clerk/astro/bundled";

Reviewer 2: I think we forgot to add the `diff` here

export default defineConfig({
    integrations: [clerk()],
    output: "server",
    adapter: node({ mode: "standalone" }),
});
```

### Changes related to stores.

> Submodule `/stores` was replaced with `/client/stores`. For the full list of the stores see [here](TODO: add reference)

Reviewer: Should we just create the doc with a title as placeholder, then we can update this link?

Author: Not sure what you mean here, do you want to drop the todo comment for now ?

Reviewer: Sorry. I mean create the `/docs/references/astro/stores` page (or maybe just a section in the Overview depending on how much content there is). Then update the link to point to that page/section.

Author: Great, that was my intention, glad you are already working on this.
